### DISCUSSION TITLE
Duplicate event message sent to SQS multiple times

### DISCUSSION THREAD

**Discussion by Set2005** [Netflix/conductor#3324] [Q&A]
Hi, it seems like same message is sent to SQS multiple times. At the beginning I thought it's the SQS visibility timeout been too short, but after having a look it seems like message is been pushed multiple times.

**Describe the issue**
Version: v3.13.2
 
1. create the workflow (included below)
2. poll for messages in AWS SQS queue
3. start the workflow from Conductor UI Workflow Workbench (Or call the Conductor API to start the workflow)
4. found two message with duplicate content

![Screen Shot 2022-10-27 at 5 48 28 am](https://user-images.githubusercontent.com/209831/198123359-309ad3f4-3675-46cb-a6c6-37dac5004216.png)

**Finding**

The recursive `decide` call at line 1040 will push the message to the queue for the second time.
conductor/core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

![Screen Shot 2022-10-27 at 5 54 00 am](https://user-images.githubusercontent.com/209831/198123685-1d1c5df3-4f47-458c-8ea0-e21624f7ac76.png)

**workflow**

```
{
  "name": "create_user",
  "description": "",
  "version": 5,
  "tasks": [
    {
      "name": "create_new_user",
      "taskReferenceName": "create_new_user",
      "inputParameters": {
        "username": "${workflow.input.username}",
        "given_name": "${workflow.input.given_name}",
        "family_name": "${workflow.input.family_name}"
      },
      "type": "EVENT",
      "startDelay": 0,
      "sink": "sqs:create_user_queue",
      "optional": false,
      "asyncComplete": false
    }
  ],
  "inputParameters": [
    "username",
    "given_name",
    "family_name"
  ],
  "outputParameters": {
    "uuid": "${create_new_user.output.uuid}",
    "email": "${create_new_user.output.email}"
  },
  "schemaVersion": 2,
  "restartable": true,
  "workflowStatusListenerEnabled": false,
  "ownerEmail": "xyz@gmail.com",
  "timeoutPolicy": "ALERT_ONLY",
  "timeoutSeconds": 3000,
  "variables": {},
  "inputTemplate": {}
}
```

**Comment by manan164** [comment]
Hi @jianbo , Thanks for reporting. We are looking into this issue and I will update the thread once we release the fix. A related issue, https://github.com/Netflix/conductor/issues/3263

**Comment by manan164** [comment]
Hi @jianbo , PR for the fix. https://github.com/Netflix/conductor/pull/3331

**Reply by jianbocui-elmo** [reply]
Thanks @manan164 , will give it a try

**Reply by jianbocui-elmo** [reply]
Issue fixed with the #3331
